File: mocks/oppgave-mock/src/main/java/no/nav/pensjon/vtp/mocks/oppgave/rest/infrastruktur/validering/IsoDateTimeValidator.java
package no.nav.pensjon.vtp.mocks.oppgave.rest.infrastruktur.validering;

import static org.springframework.util.StringUtils.isEmpty;

import javax.validation.ConstraintValidator;
import javax.validation.ConstraintValidatorContext;
import java.time.LocalDate;
import java.time.format.DateTimeFormatter;

@SuppressWarnings("WeakerAccess")
public class IsoDateTimeValidator implements ConstraintValidator<IsoDateTime, String> {

    @Override
    public void initialize(IsoDateTime isoDateTime) {
        //JSR303
    }

    @Override
    public boolean isValid(String dateTime, ConstraintValidatorContext constraintValidatorContext) {
        if (!isEmpty(dateTime)) {
            try {
                LocalDate.parse(dateTime, DateTimeFormatter.ISO_LOCAL_DATE_TIME);
            } catch (Exception e) {
                return false;
            }
        }

        return true;
    }

}

File: mocks/oppgave-mock/src/main/java/no/nav/pensjon/vtp/mocks/oppgave/rest/infrastruktur/validering/IsoDateTimeValidator.kt
package no.nav.pensjon.vtp.mocks.oppgave.rest.infrastruktur.validering

import org.springframework.util.ObjectUtils
import java.time.LocalDate
import java.time.format.DateTimeFormatter
import javax.validation.ConstraintValidator
import javax.validation.ConstraintValidatorContext

class IsoDateTimeValidator : ConstraintValidator<IsoDateTime?, String?> {
    override fun initialize(isoDateTime: IsoDateTime?) {
        // JSR303
    }

    override fun isValid(dateTime: String?, constraintValidatorContext: ConstraintValidatorContext): Boolean {
        if (!ObjectUtils.isEmpty(dateTime)) {
            try {
                LocalDate.parse(dateTime, DateTimeFormatter.ISO_LOCAL_DATE_TIME)
            } catch (e: Exception) {
                return false
            }
        }
        return true
    }
}
